src/simulator/shape_template.h

```diff
@@ -0,0 +1,59 @@
+//========================================================================
+//  This software is free: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License Version 3,
+//  as published by the Free Software Foundation.
+//
+//  This software is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  Version 3 in the file COPYING that came with this distribution.
+//  If not, see <http://www.gnu.org/licenses/>.
+//========================================================================
+/*!
+  \file    shape_template.h
+  \brief   C++ Interface: Abstract class for objects
+  \author  Yifeng Zhu, (C) 2020
+  \email   yifeng.zhu@utexas.edu
+*/
+//========================================================================
+
+#include <iostream>
+#include <vector>
+#include "eigen3/Eigen/Dense"
+#include "math/line2d.h"
+#include <cmath>
+#ifndef SHAPE_TEMPLATE_H
+#define SHAPE_TEMPLATE_H
+
+class ShapeTemplate{
+protected:
+    Eigen::Vector3f pose_; // (x, y, theta)
```

Comment:
Prefer Pose2D from the amrl_shared_lib included as a submodule of this project.

Reply:
Agreed. Please use `Pose2Df`.